### ISSUE TITLE
insights: Backfiller Improvements (backfill by series) Tracking issue

### ISSUE THREAD

**Issue by chwarwick** [sourcegraph/sourcegraph#43065]
### Plan
The original backfill process for code insights backfills all insight series runs in two steps the first step determines all the queries that need to be run for any series that needs to be backfilled and puts them in to a queue.  The 2nd step runs the queries and records the results. There are several challenges with this approach:

- Determining all the queries can be a long process and if it fails it needs to restart completely 
- The worker that runs the backfill queries processes all series in parallel which delays the time until any series is complete
- It is difficult to determine the current status of a backfill and how much work it has remaining

To solve these issues we will make a new backfilling mechanism that backfills a series at a time and breaks the work into repo sized chunks.  When backfilling it will plan, execute and record the data for 1 repo at a time. This will allow for smaller units of work that require retries if an error occurs and will enable a more accurate progress/status.  

Backfilling a series at a time means that insights over a smaller set of repos could be delayed behind insights over a larger set of repos.  To address this the backfill process will be made interruptible, so that if any work with a higher priority is created it can be processed before continuing on with lower priority work.

### Tracked issues

@chwarwick

- [ ] https://github.com/sourcegraph/sourcegraph/issues/43068 
  - [x] (🏁 86 days ago) https://github.com/sourcegraph/sourcegraph/issues/43516 

Completed
- [x] (🏁 103 days ago) https://github.com/sourcegraph/sourcegraph/issues/42431    🏳️ [Insights iteration 28](https://github.com/sourcegraph/sourcegraph/milestone/170)
- [x] (🏁 96 days ago) https://github.com/sourcegraph/sourcegraph/issues/42890 
- [x] (🏁 94 days ago) https://github.com/sourcegraph/sourcegraph/issues/42888 
- [x] (🏁 91 days ago) https://github.com/sourcegraph/sourcegraph/issues/42889 
- [x] (🏁 88 days ago) https://github.com/sourcegraph/sourcegraph/issues/42887 
- [x] (🏁 86 days ago) https://github.com/sourcegraph/sourcegraph/issues/43516 
- [x] (🏁 83 days ago) https://github.com/sourcegraph/sourcegraph/issues/42959 
- [x] (🏁 67 days ago) https://github.com/sourcegraph/sourcegraph/issues/42891 
- [x] (🏁 47 days ago) https://github.com/sourcegraph/sourcegraph/issues/43565 

@coury-clark

- [ ] https://github.com/sourcegraph/sourcegraph/issues/42328 
  - [x] (🏁 96 days ago) https://github.com/sourcegraph/sourcegraph/issues/42429    🏳️ [Insights iteration 28](https://github.com/sourcegraph/sourcegraph/milestone/170)
- [ ] https://github.com/sourcegraph/sourcegraph/issues/43068 
  - [x] (🏁 96 days ago) https://github.com/sourcegraph/sourcegraph/issues/42429    🏳️ [Insights iteration 28](https://github.com/sourcegraph/sourcegraph/milestone/170)

Completed
- [x] (🏁 96 days ago) https://github.com/sourcegraph/sourcegraph/issues/42429    🏳️ [Insights iteration 28](https://github.com/sourcegraph/sourcegraph/milestone/170)
- [x] (🏁 87 days ago) https://github.com/sourcegraph/sourcegraph/issues/43495 
- [x] (🏁 81 days ago) https://github.com/sourcegraph/sourcegraph/issues/42724    🏳️ [Insights iteration 28](https://github.com/sourcegraph/sourcegraph/milestone/170)
- [x] (🏁 77 days ago) https://github.com/sourcegraph/sourcegraph/issues/42958 
- [x] (🏁 49 days ago) https://github.com/sourcegraph/sourcegraph/issues/43890 

@leonore

Completed
- [x] (🏁 87 days ago) https://github.com/sourcegraph/sourcegraph/issues/43351 

#### Legend

- 👩 Customer issue
- 🐛 Bug
- 🧶 Technical debt
- 🎩 Quality of life
- 🛠️ [Roadmap](https://handbook.sourcegraph.com/departments/product-engineering/process/planning-process#roadmap)
- 🕵️ [Spike](https://en.wikipedia.org/wiki/Spike_(software_development))
- 🔒 Security issue
- 🙆 Stretch goal

/cc @joelkw @felixfbecker @vovakulikov

**Comment by leonore** [comment]
@chwarwick should we close this issue?
